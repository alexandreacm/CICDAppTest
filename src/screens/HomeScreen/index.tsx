import React from 'react';
import { View, StyleSheet } from 'react-native';
import dark from '../../styles/dark';
import light from '../../styles/light';
import CustomText from '../../components/CustomText';

type HomeScreenProps = {
  isDarkMode: boolean;
};

const HomeScreen: React.FC<HomeScreenProps> = ({ isDarkMode }) => {
  return (
    <View
      style={[
        styles.container,
        {
          backgroundColor: isDarkMode
            ? dark.colors.background
            : light.colors.background,
        },
      ]}
    >
      <CustomText isDarkMode={isDarkMode}>Welcome to React Native!</CustomText>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  }
});

export default HomeScreen;
