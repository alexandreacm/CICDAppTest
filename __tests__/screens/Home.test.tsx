import React from 'react';
import { render, screen } from '@testing-library/react-native';
import HomeScreen from '../../src/screens/HomeScreen';

describe('<HomeScreen />', () => {
  beforeEach(() => {
    jest.resetModules();
  });
  test('Should renders correctly',  () => {
    render(<HomeScreen isDarkMode={false} />);
  });

  test('Should renders snapshot',  () => {
    const { toJSON } = render(<HomeScreen isDarkMode={false} />);

    expect(toJSON()).toMatchSnapshot();
  });

   test('Should render Welcome text ',  () => {
    render(<HomeScreen isDarkMode={false}/>);

    screen.getByText('Welcome to React Native!');
  });

  test('Should renders correctly with dark mode',  () => {
    render(<HomeScreen isDarkMode={true} />);
  });
});
