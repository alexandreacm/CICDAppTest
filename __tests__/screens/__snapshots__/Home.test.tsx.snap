// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> Should renders snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View>
    <Text
      style={
        [
          {
            "fontSize": 20,
            "fontWeight": "bold",
          },
          {
            "color": "#000000",
          },
        ]
      }
    >
      Welcome to React Native!
    </Text>
  </View>
</View>
`;
